caikit_nlp/modules/text_generation/fine_tuning.py

```diff
@@ -282,28 +278,14 @@ def _preprocess_function(
     ):
         """Pre-process each example to get it prepared for training."""
 
-        # FIXME: Below is currently configured for Seq2Seq only
-
-        def _tokenization_func(
-            example: GenerationTrainRecord,
-        ):
-            model_inputs = tokenizer(
-                example.input,
-                max_length=max_source_length,
-                truncation=True,
-            )
-
-            labels = tokenizer(
-                example.output,
-                max_length=max_target_length,
-                padding="max_length",
-                truncation=True,
-            )
-
-            model_inputs["labels"] = labels["input_ids"]
-
-            return model_inputs
-
-        return SimpleIterableStreamWrapper(
-            train_stream.map(_tokenization_func), shuffle=shuffle
+        (
+            tokenize_function,
+            requires_unwrapping,
+        ) = base_model.build_task_tokenize_function(
+            tokenizer, max_source_length, max_target_length, verbalizer=""
         )
+        mapped_stream = train_stream.map(tokenize_function)
```

Comment:
Just a note for later, but this has the same performance pitfall we had previously discussed, which is also mentioned here https://github.com/caikit/caikit-nlp/pull/100#issuecomment-1651229995

If we revisit it in the PT module, it would be nice to handle it here also, since the solution will be the same

Reply:
true, this should easily be able to take improvements from there